test/functional/api/vim_spec.lua

```diff
@@ -2121,6 +2121,9 @@ describe('API', function()
 
       eq({}, meths.get_runtime_file("foobarlang/", true))
     end)
+    it('can handle bad patterns', function()
+      eq({}, exec_lua("return vim.api.nvim_get_runtime_file('{', true)"))
```

Comment:
Hmm this should still return (one) `Vim:E220: Missing }` error. This PR works as intended in interactive testing (proper lua exception, no repeat messages), but not here in testing for some reason.

Reply:
Yeah, this is really confusing me. It seems like the testing env is suppressing vim exceptions in some form.

FWIW this test completely hangs without the added changes.

Reply:
Debugged this and basically the error wasn't being thrown because `msg_list = NULL` for whatever reason. Using `try_enter/try_leave` uses a localized trystate which ensures `msg_list != NULL`.